src/steps/index.test.ts

```diff
@@ -39,60 +56,129 @@ test('should collect data', async () => {
+        _type: { const: 'cobalt_service' },
         _rawData: {
           type: 'array',
           items: { type: 'object' },
         },
       },
-      required: ['firstName'],
+      required: [],
     },
   });
 
-  const userGroups = context.jobState.collectedEntities.filter((e) =>
-    e._class.includes('UserGroup'),
+  const pentests = context.jobState.collectedEntities.filter((e) =>
+    e._class.includes('Assessment'),
   );
-  expect(userGroups.length).toBeGreaterThan(0);
-  expect(userGroups).toMatchGraphObjectSchema({
-    _class: ['UserGroup'],
+  expect(pentests).toMatchGraphObjectSchema({
+    _class: ['Assessment'],
     schema: {
-      additionalProperties: false,
       properties: {
-        _type: { const: 'acme_group' },
-        logoLink: {
-          type: 'string',
-          // Validate that the `logoLink` property has a URL format
-          format: 'url',
+        _type: { const: 'cobalt_pentest' },
+        _rawData: {
+          type: 'array',
+          items: { type: 'object' },
         },
+      },
+      required: [],
+    },
+  });
+
+  const findings = context.jobState.collectedEntities.filter((e) =>
+    e._class.includes('Finding'),
+  );
+  expect(findings).toMatchGraphObjectSchema({
+    _class: ['Finding'],
+    schema: {
+      properties: {
+        _type: { const: 'cobalt_finding' },
+        pentestId: { type: 'string' },
         _rawData: {
           type: 'array',
           items: { type: 'object' },
         },
       },
-      required: ['logoLink'],
+      required: ['pentestId'],
     },
   });
 });
+
+describe('testing regex extraction of URLs from pentest descriptions', () => {
+  test('recognize link and number of CWE', () => {
+    const input =
+      'Your weakness is basically https://cwe.mitre.org/data/definitions/307.html okay?';
+    expect(getVulnerabilityLink(input)).toBe(
+      'https://cwe.mitre.org/data/definitions/307.html',
+    );
+    expect(getVulnerabilityNumber(getVulnerabilityLink(input))).toBe('CWE-307');
+  });
+
+  test('recognize link and number of CVE from mitre', () => {
+    const input =
+      'Your vulnerability is basically https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-2138 okay?';
+    expect(getVulnerabilityLink(input)).toBe(
+      'https://nvd.nist.gov/vuln/detail/CVE-2021-2138', //yes, it's expected to change mitre to nvd
+    );
+    expect(getVulnerabilityNumber(getVulnerabilityLink(input))).toBe(
+      'CVE-2021-2138',
+    );
+  });
+
+  test('recognize link and number of CVE from nvd', () => {
+    const input =
+      'Your vulnerability is basically https://nvd.nist.gov/vuln/detail/CVE-2021-2138 okay?';
+    expect(getVulnerabilityLink(input)).toBe(
+      'https://nvd.nist.gov/vuln/detail/CVE-2021-2138',
+    );
+    expect(getVulnerabilityNumber(getVulnerabilityLink(input))).toBe(
+      'CVE-2021-2138',
+    );
+  });
+
+  test('recognize some link and number from a mash of links', () => {
```

Comment:
Great test documenting behavior!

In a situation like this, we'd actually want to link to both `Vulnerability` entities 😬